Number format id bug

Qwetek opened the pull request:
### Definitions
To avoid confusion (and because i dont know the oficial names) lets first define what i call "styles", "implicit number formats" and "custom number formats" in styles.xml:
Styles are xf elements inside cellXfs element inside the styleSheet.
Custom number formats are numFmt elements inside numfmts element inside styleSheet element.
Imlpicit number formats are defined by numFmtId attribute od xf element (the "style" element defined twor rows up), implicit number format ids correspond to NumberFormat.FormatNumber.

### Now for the problem:
When implicit number formats are being created when loading styles.xml from xlsx in NanoXlsX, they are getting a wrong id. Every other entity (fill, border, font...) has id equal to index, number formats do not. Number formats have special Id, non sequential. For example 0 is generic (text), 20 is timespan, 14 is date etc.. Any custom numberfromat is 164 and more. 

### Explanation:
Right now NanoXLSX assignes ids to number formats sequantially (StyleReaderContainer.GetNextNumberFormatId() which returns numberFormats.Count), so first one will get 0, second one will get 1 etc. When custom number formats are present in styles.xml, then the generated ids for implicit number formats starts at higner number (at custom number formats count). These sequential generated ids cause multiple problems:
1) When you have  at least one custom number format, and then 164 styles with attribute numFmtId = 0 (if we have more custom number formats we can have this number higher and the problem woudl still be present), then we create 164 number formats with id 1 to 164, and now we are in custom number formt ids range.
2) When you have  one custom number format, and then 13 styles with attribute numFmtId = 0 one style with numFmtId = 14, then when we are loading the style with 1 it chesk numberFormats collection for number format with id 14 and finds one, so it uses is. But the number format that was found is wrong, its for 0 (generic/text) neto for 14 (date) so our datetime is now text.
3) Even worse when we ahve the same as 2) but only 12 styles with attribute numFmtId = 0 one style with numFmtId = 20 (timespan) and one style with numFmtId = 14 (date), then our datetime style will get the existingin numberformat with id 14 which is actually the one created from 20, so our date is now timespan and if we use enforcetype in importoptions ti will be 1.1.1900

### Steps to create XlsX that will load date as timespan and when used as date in c# will produce wrong date (no extra clicking no shortcuting, no swaping steps, do exactly whats written here)
New excel (xlsx)
Set A1 text to "custom format cell" 
Set A1 format co custom a write someting thats not in implicit formats like "ss:mm"
Set A2 text to "general format with default style"
Set A3 thru A14 text to "general format with style#" where # is increasing number from 1 to 12 (texts dont matter, its just to visualise it better)
Set A3 thru A14 font colors to each different color (not default color, e need to use 12 new colros in total)
Set A15 text to "timespan" (again it does not matter what the text is)
Set A15 format exactly like this: formats => custom and select the "h:mm" (thats the one with id 20 in styles.xml) 
Set A16 text to 1.1.2024
Set A16 format to date (short) this must be the number format with id 14 in styles.xml
Save to xlsx
When loading this xlsx with NanoXlsX, cell A16 with date 1.1.2024 will be loaded as TIME and 45292.00:00:00. If we set importoptins to enforce the A column as DateTimes it will be loaded as 01.01.1900 0:00:00.
[example.xlsx](https://github.com/rabanti-github/NanoXLSX/files/14693812/example.xlsx)

rabanti-github commented on NanoXlsx Test/LowLevel/StyleReaderTest.cs:
Please move this file into the folder [NanoXlsx Test/Reader](https://github.com/rabanti-github/NanoXLSX/tree/d74eba8e4912b02f1fd972ff2d8a9bc379b2a80b/NanoXlsx%20Test/Reader)

rabanti-github commented on NanoXLSX/LowLevel/StyleReader.cs:
```diff
@@ -451,7 +451,7 @@ private void GetCellXfs(XmlNode node)
                         format = new NumberFormat();
                         format.Number = formatNumber;
-                        format.InternalID = StyleReaderContainer.GetNextNumberFormatId();
+                        format.InternalID = id;
```
In this case, we can do some cleanup, since the function GetNextNumberFormatId() is nowhere else use.
Please remove the function from the StyleContainer class.

rabanti-github reviewed (changes requested):
Hi,
First, I apologize for the long delay of this review.
Tank you very much for the PR.
There are only two things that currently should be addressed (see comments).
After that, I think everything should be fine under assumption that the writer sets the ID correctly (reader-only problem).
Thanks again

Qwetek commented:
Hello,
i believe i removed the GetNextNumberFormatId() in the initial pull request.
I moved the StyleReader tests as you requested.
S for the writer, i beieve its fine, becuase it does not generate any numberFormatIds.

rabanti-github commented:
> Hello, i believe i removed the GetNextNumberFormatId() in the initial pull request.

Sorry, your right :-) Don't know why I missed the removed code. Was clearly visible

rabanti-github commented:
This PR is now part of the new release v2.4.0. Thanks again for the contribution
